src/services/item/plugins/published/service.ts

```diff
@@ -114,6 +115,36 @@ export class ItemPublishedService {
     };
   }
 
+  private async getPublishedIfExist(item: Item, itemPublishedRepository: ItemPublishedRepository) {
+    try {
+      return await itemPublishedRepository.getForItem(item);
+    } catch (_e) {
+      return undefined;
+    }
+  }
+
+  async publishIfNotExist(
+    actor: Actor,
+    repositories: Repositories,
+    itemId: string,
+    { canBePrivate }: { canBePrivate?: boolean } = {},
+  ) {
+    if (!actor) {
+      throw new UnauthorizedMember(actor);
```

Comment:
Can we move this check into to controller ? So instead of having an Actor Object as a parameter, we have a member

Reply:
Yes it's better and it was already the case ^^.